spec/classes/hiera_spec.rb

```diff
@@ -98,12 +98,96 @@
             end
             it 'uses the provided public key path' do
               content = catalogue.resource('file', '/etc/puppet/hiera.yaml').send(:parameters)[:content]
-              expect(content).to match(%r{:pkcs7_public_key:  /path/to/public\.key})
+              expect(content).to match(%r{pkcs7_public_key: /path/to/public\.key})
             end
           end
         end
       end
+      describe 'other_backends' do
+        let(:params) do
+          {
+            merge_behavior: 'deeper',
+            eyaml: true,
+            datadir: '/etc/puppetlabs/code/environments/%{::environment}/hieradata',
+            backends: %w(yaml eyaml json yamll),
+            'backend_options' => {
+              'json' => {
+                'datadir' => '/etc/puppet/json_data/data'
+              },
+              'yamll' => {
+                'datadir' => '/etc/puppet/yamll_data/data'
+              }
+            }
+          }
+        end
+        let(:content) do
+          catalogue.resource('file', '/etc/puppet/hiera.yaml').send(:parameters)[:content]
+        end
+
+        it 'include backends' do
+          backends = YAML.load(content)['backends']
+          expect(backends).to eq(%w(yaml eyaml json yamll))
+        end
+        it 'include json backend' do
+          backend = YAML.load(content)['json']
+          expect(backend['datadir']).to eq('/etc/puppet/json_data/data')
+        end
+        it 'include yamll backend' do
+          backend = YAML.load(content)['yamll']
+          expect(backend['datadir']).to eq('/etc/puppet/yamll_data/data')
+        end
+        it 'include eyaml backend' do
```

Comment:
@bastelfreak here is where I needed the multiple expectations.  Not like its hard to write individual tests, but wanted to same compute time by checking each item individual under a single test.